spyder/plugins/explorer/tests/test_explorer.py

```diff
@@ -46,5 +66,99 @@ def test_project_explorer(project_explorer):
     assert project_explorer
 
 
+project_dir = osp.join(
+        getcwd_or_home(),
+        'temp_dir_test_file_explorer_functions_copy_save_path_file')
```

Comment:
A bit long for a dir name, don't you think?

Reply:
Yes, I agree, but these files/folders are created within Spyder repo, not in some OS temp directory. I made it long so no one in future probably will create a similar name. the folder is created in spyder/plugins/projects/widgets/tests. I slightly changed the name.

Reply:
> but these files/folders are created within Spyder repo

At least on Azure on Windows (as the failing test shows) on Windows ``get_cwd_or_home()`` returns the User home directory, not the root of the Spyder repo. Furthermore, given we're creating our own files anyway for it rather than using the existing ones is there any reason we can't just use a sudirectory of the temp dir instead? As you note the potential for conflicts and other problems makes using our repo dir quite undesirable as it leads to fragile tests and a significant potential for cross-test contamination.

> I made it long so no one in future probably will create a similar name.

Good thinking, but the better solution is to just create everything in the temp dir to avoid this and a host of other problems as discussed above. If that is impossible, ``temp_dir_test_explorer_20190104`` should be shorter and more than sufficiently unique. Besides just being longer in the source, I'm worried about edge cases on some OSes/filesystems due to length limitations and display of extremely long paths in the explorer window.

Reply:
You can use the ``tempfile`` module in the standard library to create your temporary directory, which should also take care of uniqueness and getting rid of it once you're done with it.

Reply:
The problem is that in my local testing `getcwd_or_home` function return the directory where I am running the test, which is actually the main spyder repo. I find the files created there, not in any other directory. if I use any other directory for the testing I will get again the conflict error, since getcwd_or_home will be different than temporary directory. I sure there could be a better solution, it is simply I am not aware of it.

Reply:
Right; just use ``os.chdir()`` at the beginning of the fixture to set the working directory to the temporary directory created via the ``tempfile`` module. Then, Spyder's ``getcwd_or_home()`` should return the correct, resolvable directory when used to get the relative path.

Reply:
This is certainly is resolved!